src/xrpld/app/tx/detail/Escrow.cpp

```diff
@@ -466,7 +486,16 @@ EscrowFinish::doApply()
             if (account_ != destID)
             {
                 if (!view().exists(keylet::depositPreauth(destID, account_)))
-                    return tecNO_PERMISSION;
+                {
+                    if (!ctx_.tx.isFieldPresent(sfCredentialIDs))
+                        return tecNO_PERMISSION;
+
+                    if (credentials::removeExpired(view(), ctx_.tx, j_))
+                        return tecEXPIRED;
+                    if (auto err = credentials::authorized(ctx_, destID);
+                        !isTesSuccess(err))
+                        return err;
+                }
```

Comment:
The order of operations here doesn't make sense to me. Shouldn't the expired check happen before checking for the account DP?

Reply:
fixed